### PR TITLE
Use v0.0.0 gitVersion on branches in support of new .gitattributes solution.

### PR DESCRIPTION
**Pull request by david-mcmahon**
Update bad link as well.
Prefer master branch untyped values.

ref #55353.

### CODE REVIEW

**Review by jpbetz — APPROVED** [review]
/lgtm

**Comment by ixdy** [comment]
/lgtm cancel

the verify failure is legitimate

File: pkg/version/base.go [hunk 1]
```diff
@@ -39,11 +39,11 @@ var (
 	// them irrelevant. (Next we'll take it out, which may muck with
 	// scripts consuming the kubectl version output - but most of
 	// these should be looking at gitVersion already anyways.)
-	gitMajor string = "1"  // major version, always numeric
-	gitMinor string = "8+" // minor version, numeric possibly followed by "+"
+	gitMajor = "1"  // major version, always numeric
```
Comment (ixdy) [inline pkg/version/base.go]: should we also remove these?

**Pull request merged**